internal/terraform/node_resource_abstract_instance.go

```diff
@@ -1515,9 +1512,6 @@ func (n *NodeAbstractResourceInstance) planDataSource(ctx EvalContext, currentSt
 	)
 	diags = diags.Append(checkDiags)
 	if diags.HasErrors() {
-		diags = diags.Append(ctx.Hook(func(h Hook) (HookAction, error) {
-			return h.PostApply(n.Addr, states.CurrentGen, priorVal, diags.Err())
-		}))
```

Comment:
I was a bit worried here that we'd break the timer-based "Still reading…" UI hooks here by sometimes calling `PreApply` without a corresponding `PostApply`, but after trying to trace through the code to see if that's possible I'm completely lost 😵‍💫 Just raising it in case you hadn't considered it and think it's worth looking into.

Reply:
Thanks! I could not find where this was used at all, and wasn't surprised since most of the hooks are no longer used. I didn't think about the "Still reading..." output though, so I'll walk back from there once to doublecheck before merging.

Reply:
OK, this is not even the correct place to call `PostApply`, since this section has nothing to do with reading the data source. The `PostApply` hook is still called after `ReadDataSource`, which corresponds to the ui output.